be/src/storage/task/engine_checksum_task.cpp

```diff
@@ -121,11 +120,14 @@ Status EngineChecksumTask::_compute_checksum() {
 #endif
 
         num_rows = chunk->num_rows();
-        for (auto& column : chunk->columns()) {
-            column->crc32_hash(hash_codes, 0, num_rows);
-            for (int i = 0; i < num_rows; ++i) {
-                checksum ^= hash_codes[i];
-            }
+        // The size of chunk between different replicas are not the same.
+        // So, calculate the checksum as a unit of row is necessary.
+        // TODO: MayBe there are better algorithms can calculate the checksum
+        //       instead of crc32.
+        for (int i = 0; i < num_rows; ++i) {
+            std::string row = chunk->debug_row(i);
```

Comment:
the performance is pretty low when use debug_row

Reply:
I change it column by column